### PR TITLE
Setup Testing Suite

### PR DESCRIPTION
**Pull request by skylerto**
# Summary

This PR contains a testing suite with Mocha and Chai, I also configured grunt with a default test task. 
# Tests

Test cases are located under spec/
## Strategy

The benefit of these testing frameworks are we can inject our own markdown wikis and test the generated code as front end files. Just a testing strategy worth exploring.
## Future

Move to with modularity future?
This PR contains some extra modularity.

with :heart:,
sky

### CODE REVIEW

File: lib/parse.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  let object = options || {};
+
+  if (typeof object === 'string') {
+    object = {
+      directory: object
+    };
+  }
+  if (!object.directory) {
+    // do some sort of directory finding algorithm.
```
Comment (rbren) [inline lib/parse.js]: Let's make this a TODO and throw an error for now

File: lib/parse.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  }
+  if (typeof object.bootstrap !== 'string' && object.bootstrap !== false) {
+    object.bootstrap = defaults.bootstrap;
+  }
+
+  object.marked = object.marked || defaults.marked;
+  //object.marked.gfm = true;
```
Comment (rbren) [inline lib/parse.js]: Can remove this line

File: lib/wiki.js [hunk 3]
```diff
@@ -57,3 +56,20 @@ var Wiki = module.exports = function(opts) {
   Marked.setOptions(MARKED_DEFAULTS);
 };
 
+function getOpts(opts){
```
Comment (rbren) [inline lib/wiki.js]: I don't see this function being used?

File: spec/tests.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+var chai = require('chai');
```
Comment (rbren) [inline spec/tests.js]: Is this an auto-generated file?

**Comment by rbren** [comment]
Thanks! Tests are definitely appreciated.

File: spec/tests.js (hunk 4 above)
Comment (skylerto) [inline spec/tests.js]: Oh, whoops my test setup script, that's embarrassing! Sorry about that!

File: lib/wiki.js (hunk 3 above)
Comment (skylerto) [inline lib/wiki.js]: Yikes! This function was replaced by the parser (lib/parse.js)

**Comment by rbren** [comment]
Thanks!

**Pull request merged**